### PR TITLE
NEXUS-8281: HEAD support in sender and some crucial Handlers

### PR DESCRIPTION
**Pull request by cstamas**
Issue
https://issues.sonatype.org/browse/NEXUS-8281

### CODE REVIEW

File: plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/DefaultHttpResponseSender.java [hunk 1]
```diff
@@ -48,7 +51,7 @@
     implements HttpResponseSender
 {
   @Override
-  public void send(final Response response, final HttpServletResponse httpResponse)
+  public void send(final @Nullable Request request, final Response response, final HttpServletResponse httpResponse)
```
Comment (kellyrob99) [inline plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/DefaultHttpResponseSender.java]: Minor, but technically all we need is the 'action' here, not the entire request

Comment (jdillon) [inline plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/DefaultHttpResponseSender.java]: This impl only needs that, unsure if others may or may not which is why I suggested just pass in the request and leave it up to the impl to sort out what it needs.

FTR I also think we could make this non-nullable in the future, but for now I think this is sufficient.

**Comment by jdillon** [comment]
+1

**Comment by kellyrob99** [comment]
+1

**Comment by adreghiciu** [comment]
+1

**Comment by cstamas** [comment]
http://bamboo.s/browse/NX3-OSSF112-1 :+1:

**Pull request merged**